controlplane/internal/upgrade/upgrade.go

```diff
@@ -103,3 +111,33 @@ func isKubeconfigAvailable(oacp *controlplanev1alpha2.OpenshiftAssistedControlPl
 	}
 	return kubeconfigFoundCondition.Status == corev1.ConditionTrue
 }
+
+func UpgradeWorkloadCluster(ctx context.Context, client client.Client,
+	workloadClusterClientGenerator workloadclient.ClientGenerator,
+	oacp *controlplanev1alpha2.OpenshiftAssistedControlPlane) error {
+	workloadClient, err := getWorkloadClient(ctx, client, workloadClusterClientGenerator, oacp)
+	if err != nil {
+		return err
+	}
+
+	if workloadClient == nil {
+		return fmt.Errorf("workload client is not available yet")
+	}
+
+	var clusterVersion configv1.ClusterVersion
+	if err := workloadClient.Get(ctx, types.NamespacedName{Name: "version"}, &clusterVersion); err != nil {
+		err = errors.Join(err, fmt.Errorf(("failed to get ClusterVersion from workload cluster")))
+		return err
+	}
+
+	clusterVersion.Spec.DesiredUpdate = &configv1.Update{
+		Version: oacp.Spec.DistributionVersion,
+	}
+
+	if releaseImage, ok := oacp.Annotations[UpgradeImageOverrideAnnotation]; ok {
```

Comment:
we might be able to inspect the release image and retrieve the full sha and use that instead of the annotation, I'll look into it